lib/model/autocomplete.dart

```diff
@@ -337,13 +429,25 @@ class MentionAutocompleteQuery {
 }
 
 class AutocompleteDataCache {
+  final Map<int, String> _namesByUser = {};
+
+  /// The lowercase `fullName` of [user].
+  String nameForUser(User user) {
+    return _namesByUser[user.userId] ??= user.fullName.toLowerCase();
+  }
+
   final Map<int, List<String>> _nameWordsByUser = {};
 
   List<String> nameWordsForUser(User user) {
-    return _nameWordsByUser[user.userId] ??= user.fullName.toLowerCase().split(' ');
+    final nameWords = _nameWordsByUser[user.userId];
+    if (nameWords != null) return nameWords;
+
+    final name = _namesByUser[user.userId] ??= user.fullName.toLowerCase();
```

Comment:
Let's avoid duplicating this logic:
```suggestion
    final name = nameForUser(user);
```

Particularly as we might in future put more logic there, like normalizing out diacritics.

Reply:
Then I think this method's body can be simplified back down into the form `return cache[key] ??= …initializer…;`.